### PR TITLE
Make colored-diff also work on small dataset

### PR DESCRIPTION
**Pull request by zeotuan**
Part of #136 and extension of #143 to support colored-diff on small dataset comparison

image

### CODE REVIEW

File: src/main/scala/com/github/mrpowers/spark/fast/tests/DataframeUtil.scala [hunk 1]
```diff
@@ -8,8 +8,8 @@ object DataframeUtil {
 
   private[mrpowers] def showDataframeDiff(
```
Comment (zeotuan) [inline src/main/scala/com/github/mrpowers/spark/fast/tests/DataframeUtil.scala:9]: we should probably add tests for either `showDataframeDiff` or the public assert API to make sure the displayed content are colored correctly

File: src/test/scala/com/github/mrpowers/spark/fast/tests/DatasetComparerTest.scala [hunk 2]
```diff
@@ -20,7 +21,26 @@ class DatasetComparerTest extends AnyFreeSpec with DatasetComparer with SparkSes
...
+        Person("li", 49),
+        Person("lucy", 5)
+      ).toDS
+
+      val e = intercept[DatasetContentMismatch] {
+        assertSmallDatasetEquality(sourceDS, expectedDS)
+      }
+    }
 
+    "Correctly mark unequal elements" in {
```
Comment (zeotuan) [inline src/test/scala/com/github/mrpowers/spark/fast/tests/DatasetComparerTest.scala:43]: @SemyonSinchenko , @alfonsorr - I added two small tests to validate output message of assertion. We can use ```extractColorGroup``` to add more robust testing on the message. I will create separate issue for improving tests.

**Review by SemyonSinchenko — APPROVED** [review]
LGTM

**Review by alfonsorr — APPROVED** [review]
LGTM!
One thing we have sould do is to pass scalafmt to all the project to avoid pull requests with files with changes only related with the format.

**Comment by zeotuan** [comment]
> LGTM! One thing we have sould do is to pass scalafmt to all the project to avoid pull requests with files with changes only related with the format.

Thanks! Let's add that as part of the CI pipelines. It should help with the issue

**Pull request merged**